mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -144,21 +147,32 @@ public List<InputSplit> getSplits(JobContext context) {
     InputFormatConfig.InMemoryDataModel model =
         conf.getEnum(
             InputFormatConfig.IN_MEMORY_DATA_MODEL, InputFormatConfig.InMemoryDataModel.GENERIC);
-    try (CloseableIterable<CombinedScanTask> tasksIterable = scan.planTasks()) {
-      Table serializableTable = SerializableTable.copyOf(table);
-      tasksIterable.forEach(
-          task -> {
-            if (applyResidual
-                && (model == InputFormatConfig.InMemoryDataModel.HIVE
-                    || model == InputFormatConfig.InMemoryDataModel.PIG)) {
-              // TODO: We do not support residual evaluation for HIVE and PIG in memory data model
-              // yet
-              checkResiduals(task);
-            }
-            splits.add(new IcebergSplit(serializableTable, conf, task));
-          });
-    } catch (IOException e) {
-      throw new UncheckedIOException(String.format("Failed to close table scan: %s", scan), e);
+    final ExecutorService workerPool =
+        ThreadPools.newWorkerPool(
+            "iceberg-plan-worker-pool",
+            conf.getInt(
+                SystemConfigs.WORKER_THREAD_POOL_SIZE.propertyKey(),
+                ThreadPools.WORKER_THREAD_POOL_SIZE));
+    scan = scan.planWith(workerPool);
```

Comment:
Why is this line not inside the try?

Reply:
I think scan.planWith is only an assignment and does not start a thread, but it can also be placed in try.